core/server/config/url.js

```diff
@@ -205,10 +205,12 @@ function urlFor(context, data, absolute) {
             hostname = baseUrl.split('//')[1] + ghostConfig.paths.subdir;
             if (urlPath.indexOf(hostname) > -1
                 && urlPath.indexOf('.' + hostname) === -1
-                && urlPath.indexOf('mailto:') !== 0) {
+                && urlPath.indexOf('mailto:') !== 0
+                && !urlPath.split(hostname)[0].match(/\./)) {
```

Comment:
Would probably make the code less readable, but you're right `urlPath.indexOf('.' + hostname) === -1` can be removed and 'mailto:' can be combined into one regexp.

I tried to do so `// check if there is another domain in front` but perhaps this is not clear enough? I rephrased it now, but i'm not sure if it's better now :/